_stbt/mask.py

```diff
@@ -136,6 +136,9 @@ def to_array(self, region: Region, color_channels: int = 1) \
         :returns: An image the same size as ``region``, where masked-in pixels
           are white (255) and masked-out pixels are black (0).
         """
+        if region.x != 0 or region.y != 0:
+            raise ValueError(
+                f"{region} must be a full-frame region starting at x=0, y=0")
```

Comment:
Not in this pull request. As of this pull request, you materialize the mask to a full-size numpy array by calling `to_array`, and then the caller can crop the result if you want to process a smaller region.

There are optimizations you could make to avoid allocating a full-sized frame, in some situations, but it's not trivial in all cases and isn't really worth optimizing because the result of this is memoized.